### DISCUSSION TITLE
mini.comment and .sql files?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [nvim-mini/mini.nvim#600] [Q&A] [answered]
The comments for sql (at least postgres) is `--`, but when I try to do `comment line`, `comment visual` or any other, I get this message `(mini.comment) Option 'commentstring' is empty.` and nothing happens. Using `LazyNvim` default settings, this might be vim specific, but I have no idea what to set

**Comment by echasnovski** [comment, marked as answer]
Does executing `:lua vim.bo.commentstring = '-- %s'` fix the issue?

**Reply by Nikola-Milovic** [reply]
It works!

Is there a way to make this permanent without affecting other file types?

**Reply by echasnovski** [reply]
Yes. Create a file `after/ftplugin/sql.lua` in your config directory and put line `vim.bo.commentstring = '-- %s'` in there.

**Reply by biozz** [reply]
Hi! Thank you for the tip, I've also started using it.

@echasnovski just wanted to clarify, do you think it is expected behaviour of the plugin? I think it would be great if mini.comment supported SQL files out of the box.

It would also be great if the warning from the plugin was more verbose. Also it would be great if README had a troubleshooting section related to that.

**Reply by echasnovski** [reply]
Yes, not working if 'commentstring' is not available is a completely expected behavior. 'mini.comment' [uses this option](https://github.com/echasnovski/mini.nvim/blob/df1559e2ed7ece458e0a97a8bb9556d465d1c775/doc/mini-comment.txt#L11-L14) to infer the structure of how it should comment/uncomment lines.

> It would also be great if the warning from the plugin was more verbose. Also it would be great if README had a troubleshooting section related to that.

What wording would help you in this particular case? My expectation is that 'mini.comment' is used with at least prior knowledge of what Neovim options are and how to set them. There is a note that present 'commentstring' is essential for 'mini.comment', so the wording of "Option 'commentstring' is empty" seemed OK to me at the time.

**Edit**: on Nightly (the future 0.11) this is already solved as on [2024-05-18 the 'commentstring' option](https://github.com/neovim/neovim/blob/b7e896671500a6e9a0c773bc6bac2d073e588eba/runtime/ftplugin/sql.vim#L7) is now automatically set. I also checked this on latest Nightly.

**Reply by biozz** [reply]
> What wording would help you in this particular case?

I would just put your previous answer into the readme or help text section you linked. Because it is clear and straight to the point.

```
- Comment structure by default is inferred from 'commentstring': either
  from current buffer or from locally active tree-sitter language (only on
  Neovim>=0.9). It can be customized via `options.custom_commentstring`
  (see |MiniComment.config| for details).
  Current buffer setting can also be set for the particular file type via
 `after/ftplugin/<filetype>.lua` with `vim.bo.commentstring = '-- %s'`.
```

Anyway, nice to see that it works on Nightly. I don't have any further comments on that. I should just switch to nightly as it seems the right way NeoVim community.

**Reply by echasnovski** [reply]
>   Current buffer setting can also be set for the particular file type via `after/ftplugin/<filetype>.lua` with `vim.bo.commentstring = '-- %s'`.

Yeah, I am afraid adding this type of disclaimers opens a door to many more cases like that. There needs to be a hard cutoff in what knowledge is assumed readers of help page already have.

I think having the 'commentstring' link to an option (which is assumed people know that this is a navigatable with `<C-]>` link) which says that this is a "local to buffer" option should be enough to convey all possible ways for users to interact with it.

The 'after/ftplugin/\<filetype>.lua' is a way to create a `:h ftplugin` which is a common way to set buffer-local options based on its filetype (i.e. "language", more or less). Another way is to create an autocommand for `:h FileType` event, but I prefer ftplugin approach.
